lib/passport-saml/saml.js

```diff
@@ -584,6 +592,11 @@ SAML.prototype.validatePostResponse = function (container, callback) {
     if (!Object.prototype.hasOwnProperty.call(doc, 'documentElement'))
       throw new Error('SAMLResponse is not valid base64-encoded XML');
 
+    var id = xpath(doc, "/*[local-name()='Response']/@ID");
```

Comment:
I'm fine with improving debug output, but was concerned that this `xpath()` call might add a new case that could throw. 

I checked that we get the function from `xml-crypto` which gets it from `xpath.js`, which does not document the behavior when a select fails. Turns out I opened a bug to have those docs improved in 2016, but it hasn't happened yet. 

I also confirmed there's no test coverage for that case either:

https://github.com/yaronn/xpath.js/issues/18

As long as you are improving things, could you submit a PR to `xpath.js` as well which tests and documents the behavior if a selection were to fail?

Reply:
Thanks, I'll do that and report back.

Reply:
It looks like that xpath.js project seems pretty dead, but someone has forked it, rewritten it in TypeScript and added more test coverage and docs: https://github.com/backslash47/xpath

Maybe we should convince xml-crypto to use this version instead.  

Oh, the nice new TypeScript fork depends at least Node v11 and our policy is to support the oldest LTS release of Node.js that's supported, which is currently Node.js 10. We we won't drop Node 10 support for over another year.

Reply:
Our xpath dependency changed. The correct project to be looking at is: https://github.com/goto100/xpath   It's a fork, but it retains the lack of documentation and test coverage for failing selects. At least this branch is somewhat maintained.